### ISSUE TITLE
[Feature Request] Indicate that a File is a Kanban Board in the File Explorer

### ISSUE THREAD

**Issue by phibr0** [mgmeyers/obsidian-kanban#7]
@deathau's File Extension Plugins (CSV-Editor and ini-Editor) both state in the Explorer that these files aren't default. I think it would be good to support that for Kanban's aswell. 
Of course the Files are still Markdown Files and thus don't normally have this Indicator in Obsidian.

![image](https://user-images.githubusercontent.com/59741989/115769099-7275f400-a3ab-11eb-890d-5b481a0d53d2.png)
*Example*

I think deathau "reverse engineered" the obfuscated Obsidian JS to get this method, which generates the Indicator the same as Obsidian would normally:
```js
  // this function used the regular 'document' svg,
  // but adds the supplied extension into the icon as well
  addDocumentIcon = (extension: string) => {
    addIcon(`document-${extension}`, `
  <path fill="currentColor" stroke="currentColor" d="M14,4v92h72V29.2l-0.6-0.6l-24-24L60.8,4L14,4z M18,8h40v24h24v60H18L18,8z M62,10.9L79.1,28H62V10.9z"></path>
  <text font-family="sans-serif" font-weight="bold" font-size="30" fill="currentColor" x="50%" y="60%" dominant-baseline="middle" text-anchor="middle">
    ${extension}
  </text>
    `);
```

**Comment by mgmeyers** [comment]
This would be great. I'm not sure exactly how it would be accomplished, since we only know the files are kanbans due to the frontmatter key. Worth investigating, though.

**Comment by jokysatria** [comment]
I think it's better for a different plugin. So that functionality can be used for giving any kind label to a note.

**Comment by mgmeyers** [comment]
That's a great idea @jokysatria. I'm sure a lot of folks would appreciate such a plugin.

**Comment by nothingislost** [comment]
I could probably do this by patching the ItemDom.updateTitle() method but if we're going to go down this path, we should probably use a more standard frontmatter key. Maybe something like `view-type: kanban`? Are there any other plugins doing this sort of conditional view loading based on frontmatter values?

**Comment by mgmeyers** [comment]
Yeah, good point. `file-explorer-type` seems pretty unique and possibly more understandable to users. What do you think of that?

**Comment by nothingislost** [comment]
`file-explorer-type` works for me
